hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpec.java

```diff
@@ -688,8 +699,11 @@ private void exec(@NonNull List<SpecOperation> ops) {
         final var shouldPreserveProps = !propertiesToPreserve.isEmpty();
         final Map<String, String> preservedProperties = shouldPreserveProps ? new HashMap<>() : Collections.emptyMap();
         if (shouldPreserveProps) {
-            ops = new ArrayList<>(ops);
-            ops.add(0, remembering(preservedProperties, propertiesToPreserve));
+            ops.addFirst(remembering(preservedProperties, propertiesToPreserve));
+        }
+        if (throttleResource != null) {
+            ops.addFirst(new SysFileOverrideOp(
+                    THROTTLES, () -> throttleResource.isBlank() ? null : resourceAsString(throttleResource)));
```

Comment:
can you move the predicate `throttleResource.isBlank()` up to the enclosing `if` at L704?

Reply:
The goal is to only add a `SysFileOverrideOp` if the `HapiSpec#throttleResource` field is non-null.

When we **do** have a non-null `throttleResource`, if it is not blank, then we will automatically override file `0.0.123` with the parsed contents of the given resource. If it is blank, we will just take a snapshot of its initial contents and restore them at the end of the spec.